packages/frontend-2/components/server-management/SettingsDialog.vue

```diff
@@ -0,0 +1,185 @@
+<template>
+  <LayoutDialog
+    v-model:open="isOpen"
+    max-width="sm"
+    :title="title"
+    :buttons="dialogButtons"
+  >
+    <form @submit="onSubmit">
+      <div class="flex flex-col gap-6">
+        <FormTextInput
+          v-model="name"
+          label="This server’s public name"
+          name="serverName"
+          placeholder="Server name"
+          show-label
+          :show-required="true"
+          :rules="requiredRule"
+          :type="'text'"
+        />
+        <FormTextArea
+          v-model="description"
+          label="Description"
+          name="description"
+          placeholder="Description"
+          show-label
+        />
+        <FormTextInput
+          v-model="company"
+          label="Owner"
+          name="owner"
+          placeholder="Owner"
+          show-label
+        />
+        <FormTextInput
+          v-model="adminContact"
+          label="Admin Email"
+          name="adminEmail"
+          placeholder="Admin Email"
+          show-label
+          :rules="emailRules"
```

Comment:
Currently this prevents you from leaving the field empty or typing in something like "n/a"